test/partial_prox.jl

```diff
@@ -54,5 +54,20 @@ for op ∈ (:NormL0, :NormL1, :RootNormLhalf)
         end
       end
     end
+
+    # tests iprox without bounds
+    if op == :NormL0 || op == :NormL1
+      for d ∈ [ones(n), 2 * ones(n), zeros(n)]
```

Comment:
Hi @MohamedLaghdafHABIBOULLAH , this raises an error with your `iprox!` when `d = zeros(n)`, you should remove it here.
You can test the error in the unit tests when `d = zeros(n)` using this for example:
https://docs.julialang.org/en/v1/stdlib/Test/#Test.@test_throws